### PR TITLE
M3-5204: Add Cost Estimator for Custom Images

### PR DESCRIPTION
**Pull request by codyfinn**
## Description

Adds price estimation for creating custom images from existing Linode Disks. The price estimation is based on the size of the disk and the current unit cost of $0.10 / GB of Disk storage. Included is also utility function to convert between storage units easily and another utility function to calculate total prices given a unit price and a given quantity.

## How to test

Navigate to the CreateCustomImageTab component and select a Linode and a Disk. The cost estimator should show the correct price for the size of the disk.

**How do I run relevant unit tests?**

`yarn test unitPriceCalculator`
`yarn test unitConversions --t convertStorageUnit`

### CODE REVIEW

**Review by johnwcallahan — COMMENTED** [review]
UI question for Matthew and the team: should we consider adding "Estimated: " before the price? (since you could e.g. resize the disk).

File: packages/manager/src/utilities/unitConversions.ts [hunk 1]
```diff
@@ -189,3 +189,34 @@ export const convertBytesToTarget = (unit: StorageSymbol, value: number) => {
...
+  TB = 4,
+}
+
+type StorageUnitExponentKey = keyof typeof StorageUnitExponents;
+
+export const convertStorageUnit = (
+  sourceUnit: StorageUnitExponentKey,
+  sourceQuantity: number | undefined,
+  targetUnit: StorageUnitExponentKey
+) => {
```
Comment (bnussman) [inline packages/manager/src/utilities/unitConversions.ts:207]: Can you add a tsdoc style comment for this function?

File: packages/manager/src/utilities/unitPriceCalculator.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function unitPriceCalculator(
+  unitPrice: number,
+  quantity: number
+) {
+  return `$${Math.max(unitPrice * quantity, 0).toFixed(2)}`;
+}
```
Comment (bnussman) [inline packages/manager/src/utilities/unitPriceCalculator.ts:6]: And maybe this one if you feel like its would be useful to readers

**Review by DevDW — COMMENTED** [review]
Looking good to me so far

File: packages/manager/src/utilities/unitConversions.ts [hunk 3]
```diff
@@ -176,16 +176,48 @@ const determineDecimalPlaces = (
 export const convertBytesToTarget = (unit: StorageSymbol, value: number) => {
   switch (unit) {
     case 'byte':
-      return value;
     case 'bytes':
+    case 'B':
```
Comment (DevDW) [inline packages/manager/src/utilities/unitConversions.ts:180]: Seeing this TS error:
Screen Shot 2021-08-19 at 11 10 08 AM

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx [hunk 4]
```diff
@@ -162,6 +164,15 @@ export const CreateImageTab: React.FC<Props & ImagesDispatch> = (props) => {
 
   const requirementsMet = checkRequirements();
 
+  const selectedDiskData: Disk | undefined = disks.find(
+    (d) => `${d.id}` === selectedDisk
+  );
+  const selectedDiskSizeInMB = convertStorageUnit(
```
Comment (DevDW) [inline packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:170]: I think this variable should be renamed to `selectedDiskSizeInGB`

**Review by DevDW — APPROVED** [review]
Nice work! Left a couple of small comments

**Review by bnussman — APPROVED** [review]
🚀

**Pull request merged**